### PR TITLE
feat: refresh button

### PR DESCRIPTION
**Pull request by pbullhove**
## What does this pull request change?

Adding refresh button

### CODE REVIEW

**Comment by henrikbossart** [comment]
Looks good, need to support configuring the button in the entity though.
Only gripe I have is that is looks slightly scuffed with the lack of space/padding between the border and the content when hovering. Not quite sure how to fix that, as it would need some padding on the content. And I don't think that works for all cases

File: packages/dm-core/src/components/EntityView.tsx [hunk 1]
```diff
@@ -28,13 +37,29 @@ export const EntityView = (props: IEntityView): React.ReactElement => {
     dimensions
   )
 
+  // Refresh Button stuff
+  const [reloadCounter, setReloadCounter] = useState(0)
+  const [mouseHoverOverRefreshButton, setMouseHoverOverRefreshButton] =
+    useState(false)
+  const [mouseCloseToRefreshButton, setMouseCloseToRefreshButton] =
+    useState(false)
```
Comment (soofstad) [inline packages/dm-core/src/components/EntityView.tsx:45]: Cannot this hover effect be more elegantly be handled with CSS?
I think so... :)

File: packages/dm-core/src/components/EntityView.tsx [hunk 2]
```diff
@@ -28,13 +37,29 @@ export const EntityView = (props: IEntityView): React.ReactElement => {
...
+  const reloadComponent = () => setReloadCounter(reloadCounter + 1)
```
Comment (soofstad) [inline packages/dm-core/src/components/EntityView.tsx:46]: Not a big fan of these functions that only call another function. IMO it's better to call "setReloadCounter" directly.
Helps with readability

**Review by soofstad — CHANGES_REQUESTED** [review]

**Comment by soofstad** [comment]
Looks good. Works great.
Think we can wait with making it configurable.
Also, move the button just a few pixles left and down, so that it does not overlap with the border?
And, is there something we can do to make it more discoverable. It's quite hard to find the button in some of the plugins/grids/layout. Perhaps make the icon itself always be visible, with high opacity. Experiment and see what works :shrug:

**Comment by henrikbossart** [comment]
>  Perhaps make the icon itself always be visible, with high opacity.

I assume 'low opacity' is meant?

File: packages/dm-core/src/components/EntityView.tsx (hunk 1 above)
Comment (pbullhove) [reply packages/dm-core/src/components/EntityView.tsx:45]: Hmm how so?

By wrapping it in a invisible container of 1/3 size and having the mouseEnter and mouseLeave property here?

File: packages/dm-core/src/components/EntityView.tsx (hunk 2 above)
Comment (pbullhove) [reply packages/dm-core/src/components/EntityView.tsx:46]: alright

File: packages/dm-core/src/components/EntityView.tsx (hunk 1 above)
Comment (henrikbossart) [reply packages/dm-core/src/components/EntityView.tsx:45]: Could probably make the button opacity work with only css, but I don't think you can style different elements based on hovering another with pure CSS 🤔

**Comment by pbullhove** [comment]
> Looks good, need to support configuring the button in the entity though. Only gripe I have is that is looks slightly scuffed with the lack of space/padding between the border and the content when hovering. Not quite sure how to fix that, as it would need some padding on the content. And I don't think that works for all cases

Managed to fix this with 'outline' instead of 'border' which does not _shift_ the content inside, and by adding a property 'outlineOffset'.

File: packages/dm-core/src/components/EntityView.tsx (hunk 1 above)
Comment (soofstad) [reply packages/dm-core/src/components/EntityView.tsx:45]: you can set different styling based on "hover" and other similar properties like so:
```
const ClickableIcon = styled.button`
  appearance: none;
  border: none;
  background-color: transparent;

  &:hover {
    color: gray;
    cursor: pointer;
  }
`
```

**Comment by henrikbossart** [comment]
> > Looks good, need to support configuring the button in the entity though. Only gripe I have is that is looks slightly scuffed with the lack of space/padding between the border and the content when hovering. Not quite sure how to fix that, as it would need some padding on the content. And I don't think that works for all cases
> 
> Managed to fix this with 'outline' instead of 'border' which does not _shift_ the content inside, and by adding a property 'outlineOffset'.

Right, but doesn't the outline give a rather ugly "edgy" outline? 🤔

**Comment by pbullhove** [comment]
> > > Looks good, need to support configuring the button in the entity though. Only gripe I have is that is looks slightly scuffed with the lack of space/padding between the border and the content when hovering. Not quite sure how to fix that, as it would need some padding on the content. And I don't think that works for all cases
> > 
> > 
> > Managed to fix this with 'outline' instead of 'border' which does not _shift_ the content inside, and by adding a property 'outlineOffset'.
> 
> Right, but doesn't the outline give a rather ugly "edgy" outline? 🤔 Or does it curve smoothly like the border?

Can have border-radius on the outline as well.

File: example/app/data/DemoDataSource/recipes/plugins/grid/car_grid/car.recipe.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "_blueprintPath_": "/plugins/grid/car_grid/blueprints/Car",
   "initialUiRecipe": {
     "name": "form",
-    "type": "CORE:UiRecipe",
+    "type": "CORE:git ",
```
Comment (henrikbossart) [inline example/app/data/DemoDataSource/recipes/plugins/grid/car_grid/car.recipe.json:6]: this seems like an unintended change?

Comment (pbullhove) [reply example/app/data/DemoDataSource/recipes/plugins/grid/car_grid/car.recipe.json:6]: hhehe nice catch

**Review by soofstad — APPROVED** [review]
Looks great!

**Pull request merged**